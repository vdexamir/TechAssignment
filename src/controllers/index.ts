import { MockCalendarEventRepository } from "../repositories";
import { CalendarApiService } from "../services/CalendarApiService";
import { CalendarApiController } from "./CalendarApiController";

const deps = {
  repository: new MockCalendarEventRepository(),
};
const service = new CalendarApiService(deps);

export const calendarApi = new CalendarApiController(service);
