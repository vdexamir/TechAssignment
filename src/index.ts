export * from "./core/domain";
export {
  BadRequestError,
  NotFoundError,
  ConflictError,
  InternalServerError,
  HttpStatus,
} from "./core/errors";
export * from "./core/services";
export * from "./controllers";
