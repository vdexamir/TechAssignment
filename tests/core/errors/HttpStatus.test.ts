import { HttpStatus } from "../../../src";
import { describe, expect, it } from "@jest/globals";

describe("HttpStatus", () => {
  it("should match the snapshot", () => {
    expect(HttpStatus).toMatchSnapshot();
  });
});
