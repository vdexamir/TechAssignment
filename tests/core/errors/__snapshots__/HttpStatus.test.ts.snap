// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpStatus should match the snapshot 1`] = `
{
  "200": "ok",
  "201": "created",
  "400": "badRequest",
  "401": "unauthorized",
  "403": "forbidden",
  "404": "notFound",
  "409": "conflict",
  "500": "internalServerError",
  "badRequest": 400,
  "conflict": 409,
  "created": 201,
  "forbidden": 403,
  "internalServerError": 500,
  "notFound": 404,
  "ok": 200,
  "unauthorized": 401,
}
`;
